Sara/tec 147 october 2024 release notes (#1810)
semgrep/semgrep-docs: docs/release-notes/october-2024.md

---
slug: october-2024
title: October 2024
hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
tags:
  - Release notes
---

# Semgrep release notes for October 2024

## 🌐 Semgrep AppSec Platform

### Added

> - Added a **Scan details** page and drawer for all completed scans. Use this page to troubleshoot or view information about individual scans. 

Reviewer: Do we want to explain what we mean by drawer?

Author: I'll check our trusty google style guide if it has a preferred or more common term!

Author: They prefer calling it a notification drawer, which doesn't match our use case.
https://developers.google.com/style/word-list#notification-drawer

I'll stick to calling it a sliding pane: https://developers.google.com/style/ui-elements

_**Figure**. Scan details drawer with the permalink icon indicated in a box._
- The **Dashboard** now provides a **Teams** filter, enabling you to create views based on a selection of [Teams](/deployment/teams#teams-beta) you are a part of. Click **Dashboard > Filters** to access the filter.
  - By default, the Dashboard now displays findings from teams you are a part of. Your finding count may differ from your colleagues based on your Teams.
- Added a Jira API endpoint to create Jira tickets, either by passing a list of `issue_ids` or filter query parameters to select findings. Refer to the [<i class="fas fa-external-link fa-xs"></i> Jira API documentation](https://semgrep.dev/api/v1/docs/#tag/TicketingService/operation/semgrep_app.core_exp.notifications.ticketing.handlers.openapi_create_tickets).
- Semgrep now supports [Move on Sui](https://docs.sui.io/concepts/sui-move-concepts), thanks to the contributions of the Sui team.

### Changed

- Various UI improvements to the **Settings > SCM** tab.
![Old SCM card](/img/old-scm-card.png) 
![Updated SCM card](/img/new-scm-card.png)
_**Figure**. Previous and current SCM card UI._

> - **Semgrep Managed Scans**: scans now follow fail open behavior, consistent with how Semgrep in CI behaves. 

Reviewer: Maybe define what fail open means?

> - The **Projects** page's **See findings** button is now a drop-down box, enabling you to select which product you want to view findings for.

Reviewer: ```suggestion
- The **Project details** page's **See findings** button is now a drop-down box, enabling you to select which product you want to view findings for.
```

Author: Thank you for this catch!

### Fixed

> - When a scan runs into an exception, Semgrep AppSec Platform is appropriately notified about the failure. Previously, within the AppSec Platform UI, it would appear to the user that the scan is still in progress.

Reviewer: ```suggestion
- When a scan runs into an exception, Semgrep AppSec Platform displays information about the failure. Previously, within the AppSec Platform UI, it would appear to the user that the scan is still in progress.
```

- Fixed a bug where Semgrep would crash if `--trace` was passed.

## 💻 Semgrep Code

### Added

- Updated the C# parser to support all versions of the language up to 13.0 (.NET 9).
- Developers can now triage findings by replying to a GitHub PR comment from Semgrep, without the need to log in to Semgrep Cloud Platform. See [Triage findings through comments](/semgrep-code/triage-remediation#triage-findings-through-pr-and-mr-comments) for more information.
- Added an API endpoint you can use to triage findings in bulk, either by passing a list of `issue_ids` or filter query parameters to select findings. Refer to [<i class="fas fa-external-link fa-xs"></i> Bulk triage API documentation](https://semgrep.dev/api/v1/docs/#tag/TriageService).
- Taint analysis now supports tracking sinks through callbacks for all applicable Semgrep-supported languages. For example:

> ```javascript

Reviewer: Indent this code block so it lines up under the bullet point

function unsafe_callback(x) {
  sink(x); // Semgrep detects a finding here now!
}

function withCallback(val, callback) {
  callback(val);
}

withCallback(taint, unsafe_callback)
```

### Removed

- Removed support for Vue. The `tree-sitter grammar` has not been updated in 3 years and no community rules have been added. In theory, extract mode could be a good substitute to parse Vue files.

## ⛓️ Semgrep Supply Chain

### Added

- Supply Chain now provides reachability analysis for Kotlin, including support for Gradle and Maven.
- Improved support and flexibility to Python dependency parsing (public beta):
  - Semgrep now finds non-standard `requirements.txt` names and parses them for dependencies. 
  - Semgrep parses lockfiles in a `requirements` folder.
- `cargo.lock` parser can now associate dependencies with lockfile line numbers.

### Changed

- Improvements to the **Advisories** page UI. <!-- 16657 -->

> - **Dependency search**: The **Ecosystem** filter has been replaced by a **Language** filter. Several languages can share the same ecosystem, such as Java and Kotlin both using Maven. For accurate filtering, the Dependencies page now uses a Language filter so that you can view that language's packages from any ecosystem supported by Semgrep for that language.

Reviewer: ```suggestion
- **Dependency search**: The **Ecosystem** filter has been replaced by a **Language** filter. Several languages can share the same ecosystem, such as Java and Kotlin both using Maven. For accurate filtering, the **Dependencies** page now uses a **Language** filter so that you can view that language's packages from any ecosystem supported by Semgrep for that language.
```

### Fixed

- **Advisories** page: improved speed when fetching advisories.

## 🤖 Semgrep Assistant

### Added

- Users can now use Semgrep Assistant with their own OpenAI API key.
  - Enterprise users can also use the following API providers:
    - Azure OpenAI
    - AWS Bedrock
    - Google Gemini

>  - See the [AI provider documentation](/semgrep-assistant/getting-started#use-your-ai-provider) for more details.

Reviewer: ```suggestion
 See the [AI provider documentation](/semgrep-assistant/getting-started#use-your-ai-provider) for more details.
```
I don't know if this suggestion will work, but I don't think this needs a bullet point; it just needs to be nested accordinly.

- PR comments made by Semgrep Assistant now reference the Git commits that it used to generate the fix. <!-- 17152 -->
![Semgrep Assistant referencing multiple commits](/img/semgrep-assistant-reference-commits.png)
_**Figure**. Semgrep Assistant referencing multiple commits._

## 🔐 Semgrep Secrets

> - `semgrep ci` output now includes a list of all secrets rules which generated at least one blocking finding, consistent with Semgrep Code behavior.

Reviewer: ```suggestion
- `semgrep ci` output now includes a list of all Secrets rules which generated at least one blocking finding. This behavior is consistent with that of Semgrep Code.
```

## 📝 Documentation and knowledge base

### Added

- Documented new triage workflows.
- Improvements to the **[Network broker documentation](/semgrep-ci/network-broker)**.
- Updated [Supported languages](/supported-languages) with new languages and features.

> - Added new sections in Semgrep AppSec Platform vs Semgrep OSS.

Reviewer: I think we should either link to all the docs or have no links at all. WDYT?
